### PR TITLE
Create a default organization member object for superusers when no member exists in the database.

### PR DESCRIPTION
**Pull request by Donnype**
### Changes
Create a default organization member object for superusers when no member exists in the database.

### Issue link
Closes https://github.com/minvws/nl-kat-coordination/issues/1498

### Proof
Accessing a newly created organization with a different superuser than the one that created it:
image

---

### Code Checklist
- [x] All the commits in this PR are properly PGP-signed and verified;
- [x] This PR only contains functionality relevant to the issue; tickets have been created for newly discovered issues.
- [x] I have written unit tests for the changes or fixes I made.
- [x] For any non-trivial functionality, I have added integration and/or end-to-end tests.
- [x] I have performed a self-review of my code and refactored it to the best of my abilities.

### Communication
- [x] I have informed others of any required `.env` changes files if required and changed the `.env-dist` accordingly.
- [x] I have made corresponding changes to the documentation, if necessary.

---
## Checklist for code reviewers:
Copy-paste the checklist from [the docs/source/templates folder](https://github.com/minvws/nl-kat-coordination/blob/main/docs/source/templates/pull_request_template_review_code.md) into your comment.

---
## Checklist for QA:
Copy-paste the checklist from [the docs/source/templates folder](https://github.com/minvws/nl-kat-coordination/blob/main/docs/source/templates/pull_request_template_review_qa.md) into your comment.

### CODE REVIEW

**Comment by underdarknl** [comment]
Looks good to me, To answer your question in the bug for situation #2. if a superuser decided to change their organisationMember role somehow, I would find it logical for that to work and create a database stored record with those new settings, which would be used on subsequent visits.
I would find it more clean if  Organisation() or OrganisationMember() methods would reside on the User Model and would function with the User object as their reference point instead of the request.user as we now do in the view.
Maybe this would be a refactoring step we can pick up in a separate issue.

File: rocky/account/mixins.py [hunk 1]
```diff
@@ -89,7 +89,16 @@ def setup(self, request, *args, **kwargs):
...
-            raise Http404()
+            if not self.request.user.is_superuser:
+                raise Http404()
+
+            self.organization_member = OrganizationMember(
+                user=self.request.user,
+                organization=self.organization,
+                status=OrganizationMember.STATUSES.ACTIVE,
+                trusted_clearance_level=4,
+                acknowledged_clearance_level=4,
```
Comment (praseodym) [inline rocky/account/mixins.py:100]: Maybe the clearance level shouldn't be acknowledged by default?

Comment (Donnype) [reply rocky/account/mixins.py:100]: Perhaps, although this is also the default now for organisation members that get created for superusers when they add an organisation:
Schermafbeelding 2023-08-31 om 14 36 01

Comment (praseodym) [reply rocky/account/mixins.py:100]: One could argue that creating an organisation yourself gives you different permissions than joining an existing one, but maybe we should also remove the acknowledged clearance level from `views/organization_add.py`?

Comment (Donnype) [reply rocky/account/mixins.py:100]: Yes, the minimal extra steps superusers then need to take also make sure they're aware they're dealing with a new organization. If @underdarknl agrees I can set the default `acknowledged_clearance_level` to 0 in both scenario's.

**Review by praseodym — APPROVED** [review]
lgtm

File: rocky/rocky/locale/pap/LC_MESSAGES/django.po [hunk 2]
```diff
@@ -1846,15 +1846,20 @@ msgstr ""
...
-"autorisashon di L%s. Tuma kontakto ku bo administrator pa risibí un nivel di "
-"autorisashon mas haltu."
+
+#: katalogus/views/mixins.py onboarding/views.py
+#, python-format
+msgid ""
+"Could not raise clearance level of %s to L%s. You acknowledged a clearance "
+"level of L%s. Please accept the clearance level first on your profile page "
+"to proceed."
+msgstr ""
```
Comment (Rieven) [inline rocky/rocky/locale/pap/LC_MESSAGES/django.po:1862]:
```suggestion
msgstr "No por a subi na e nivel di autorisashon di %s pa L%s. Bo a rekonosé un nivel di autorisashon di L%s. Por fabor, akseptá e nivel di autorisashon promé riba bo página di perfil pa por sigui."
```

File: rocky/rocky/locale/pap/LC_MESSAGES/django.po [hunk 3]
```diff
@@ -5409,12 +5405,16 @@ msgstr ""
...
-"No por subi e nivel di autorisashon pa L%s. Bo a rekonosé un nivel di "
-"autorisashon di L%s. Tuma kontakto ku bo administrator pa risibí un nivel di "
-"autorisashon mas haltu."
+
+#: rocky/views/ooi_list.py
+#, python-format
+msgid ""
+"Could not raise clearance level to L%s. You acknowledged a clearance level "
+"of L%s. Please accept the clearance level below to proceed."
+msgstr ""
```
Comment (Rieven) [inline rocky/rocky/locale/pap/LC_MESSAGES/django.po:5417]:
```suggestion
msgstr "No por a subi na e nivel di autorisashon pa L%s. Bo a rekonosé un nivel di autorisashon di L%s. Por fabor, akseptá e nivel di autorisashon abou pa por sigui."
```

File: rocky/rocky/locale/pap/LC_MESSAGES/django.po [hunk 4]
```diff
@@ -5627,6 +5627,13 @@ msgstr ""
 "No por a subi e nivel di autorisashon di %s pa L%s. Indemnisashon no ta "
 "presente pa organisashon %s."
 
+#: rocky/views/scan_profile.py
+#, python-format
+msgid ""
+"Could not raise clearance level of %s to L%s. You acknowledged a clearance "
+"level of L%s. Please accept the clearance level below to proceed."
+msgstr ""
```
Comment (Rieven) [inline rocky/rocky/locale/pap/LC_MESSAGES/django.po:5635]:
```suggestion
msgstr "No por a subi e nivel di autorisashon di %s pa L%s. Bo a rekonosé un nivel di autorisashon di L%s. Por fabor, akseptá e nivel di autorisashon mas abou pa por sigui."
```

File: rocky/account/mixins.py (hunk 1 above)
Comment (TwistMeister) [reply rocky/account/mixins.py]: I think this depends on ones understanding/interpretation of what a superuser is or should be. Afaik someone who is superuser should be able to access all organisations and should be able to perform all actions. I have no real reference for this, but I think that it is why a superuser gets an acknowledged level of 4 by default.

**Review by TwistMeister — APPROVED** [review]

**Pull request merged**